packages/datadog-plugin-confluentinc-kafka-javascript/test/index.spec.js

```diff
@@ -332,23 +341,59 @@ describe('Plugin', () => {
           })
 
           describe('consumer', () => {
-            beforeEach(() => {
+            beforeEach(async () => {
               nativeConsumer = new Consumer({
                 'bootstrap.servers': '127.0.0.1:9092',
-                'group.id': 'test-group-native'
+                'group.id': 'test-group'
               })
 
-              nativeConsumer.on('ready', () => {
-                nativeConsumer.subscribe([testTopic])
+              await new Promise((resolve, reject) => {
+                nativeConsumer.connect({}, (err) => {
+                  if (err) {
+                    return reject(err)
+                  }
+                  resolve()
+                })
               })
-
-              nativeConsumer.connect()
             })
 
-            afterEach(() => {
-              nativeConsumer.disconnect()
+            afterEach(async () => {
+              await nativeConsumer.unsubscribe()
+              await new Promise((resolve, reject) => {
+                nativeConsumer.disconnect((err) => {
+                  if (err) {
+                    return reject(err)
+                  }
+                  resolve()
+                })
+              })
             })
 
+            function consume (consumer, producer, topic, message) {
+              return new Promise((resolve, reject) => {
+                function doConsume () {
+                  consumer.consume(1, function (err, messages) {
```

Comment:
I assume we expect this error to never be present and if it is, the test should fail? We might not expect that in this test, but since this PR is to reduce flakiness, maybe it's a good idea to check?

```suggestion
                  consumer.consume(1, function (err, messages) {
                    if (err) return reject(err)
```